### PR TITLE
Use sharedPodInformer in ReplicaSet controller

### PR DESCRIPTION
**Pull request by wojtek-t**
@caesarxuchao

### CODE REVIEW

**Comment by wojtek-t** [comment]
@k8s-bot e2e test this please, issue: #IGNORE (cluster didn't started due to infra problems)

**Comment by wojtek-t** [comment]
@k8s-bot e2e test this, issue: #IGNORE (fixed by https://github.com/kubernetes/kubernetes/pull/29278)

**Comment by kargakis** [comment]
Make sure the replica set is deep-copied before any mutation (if it's not the case already)

**Comment by kargakis** [comment]
> Make sure the replica set is deep-copied before any mutation (if it's not the case already)

Oh, you are sharing only the pod informer. Last time I checked there wasn't any pod mutation happening but make sure just in case.

**Comment by wojtek-t** [comment]
@kargakis - I went through the code and it seems save to me (actually, this is basically the copy of replication controller, and this is already done there).

**Comment by caesarxuchao** [comment]
LGTM. Since this might be disruptive, I'll give @deads2k a chance to take another pass.

File: pkg/controller/replicaset/replica_set.go [hunk 1]
```diff
@@ -84,7 +92,7 @@ type ReplicaSetController struct {
 	// A store of pods, populated by the podController
 	podStore cache.StoreToPodLister
 	// Watches changes to all pods
-	podController *framework.Controller
+	podController framework.ControllerInterface
```
Comment (deads2k) [inline pkg/controller/replicaset/replica_set.go]: Is this used for anything besides `HasSynced`?  If not, how about just pulling that directly from the informer?

**Comment by deads2k** [comment]
nit, lgtm otherwise.

File: pkg/controller/replicaset/replica_set.go (hunk 1 above)
Comment (wojtek-t) [inline pkg/controller/replicaset/replica_set.go]: I copied it from ReplicationController which is mostly the same code as in ReplicaSet here.

I can remove it, but if you are ok with it, I would prefer doing this in the next PR, which this PR is blocking.
Does it sound ok?

Comment (deads2k) [inline pkg/controller/replicaset/replica_set.go]: > I can remove it, but if you are ok with it, I would prefer doing this in the next PR, which this PR is blocking.
> Does it sound ok?

Sure.

**Comment by wojtek-t** [comment]
Adding label based on above comments from both @deads2k and @caesarxuchao 

Thanks!

**Pull request merged**